### ISSUE TITLE
Intermittent Numeric Tests Failures on OSX

### ISSUE THREAD

**Issue by ellismg** [dotnet/corefx#5511]
If you look a the history for the [OSX Release Test Job](http://dotnet-ci.cloudapp.net/job/dotnet_corefx/job/osx_release_tst/) some of the matrix tests have been failing intermittently.  I'm not sure if the failures are machine specific (one of the Matrix 4x4 tests has failed twice on -1 mac but passes elsewhere).

**Comment by mellinoe** [comment]
I haven't been able to repro this on my machine. This is related to #4882 , it seems. By the way, these tests are deterministic, so I don't really understand why they are only failing occasionally when each run should be identical.

**Comment by ellismg** [comment]
@mellinoe I am wondering if it passes on some machines but fails on others?  Would the code we run differ based on what sort of machine we are running on?

**Comment by mellinoe** [comment]
As far as I know it shouldn't differ across machines, especially not on OSX, because SIMD isn't enabled on OSX yet

For other context, these same tests have been running unchanged for a long time, on .NET Native, Desktop, CoreCLR, etc. So I strongly suspect there's a bug in the (OSX) JIT somewhere, but I don't know where. The fact that it is an unreliable repro also makes it very hard to diagnose. I filed https://github.com/dotnet/coreclr/issues/2266, but we weren't able to repro the issue there. There were some JIT fixes at the time so we assumed it was fixed. I'll re-open that as it's still occurring.
